include/triton/Dialect/Triton/IR/TritonOps.td

```diff
@@ -498,13 +497,18 @@ def TT_MakeRangeOp : TT_Op<"make_range", [Pure]> {
         Values span from $start to $end (exclusive), with step = 1
     }];
 
+    // WARNING: MLIR generates getStart()/getEnd() functions which return
+    // uint32_t, even though these arguments are to be interpreted as *signed*
+    // int32 values.  If this matters, use get{Start,End}Attr().getInt(), which
+    // returns int64_t.
```

Comment:
nit: Should we use `SI32Attr`? Not sure if this would have other side effects

Reply:
You mean changing the attribute type to SI32Attr?  If we did this, we'd have to update all the tests and IR generation to use si32 rather than i32, I think?

Perhaps worth doing, but we probably have bigger fish to fry.  I did audit all uses of getStart and getEnd and they looked correct.

Reply:
yes I meant using `SI32Attr`, si32 is for Values right? Aren't the two independent? Attributes are always literals. I agree if it is not as trivial as just replace `I32Attr` with `SI32Attr` it is not worth it.

Reply:
Tried it, and yeah, it looks like you'd have to change the textual IR of every test that uses MakeRange (probably plus all the C++ code).

```
error: unexpected error: custom op 'tt.make_range' 'tt.make_range' op attribute 'end' failed to satisfy constraint: 32-bit signed integer attribute
    %a = tt.make_range { start = 0 : i32, end = 32 : i32 } : tensor<16xi32>
```

Reply:
ah ok, that sucks, alright that was a bad idea